qemu/tests/driver_update.py

```diff
@@ -0,0 +1,148 @@
+import os
+
+from autotest.client.shared import error
+from autotest.client.shared import utils
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import data_dir
+from virttest import env_process
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Update virtio driver:
+    1) Boot up guest with default devices and virtio_win iso
+    2) Install virtio driver
+    3) Check dirver info
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+
+    def reboot(vm, session=None):
+        nic_idx = len(vm.virtnet) - 1
+        while nic_idx >= 0:
+            try:
+                return vm.reboot(session, nic_index=nic_idx)
+            except Exception:
+                nic_idx -= 1
+                if nic_idx < 0:
+                    raise
+                logging.warn("Unable to login guest, "
+                             "try to login via nic %d" % nic_idx)
+
+    def get_installation_log(vm, session=None):
+        """
+        Copy driver debug files DPINST.log and setupapi.dev.log
+        from guest to host
+        """
+        error.context("Copy installation log from guest to host", logging.info)
+        debug_files = "C:\Windows\DPINST.log,C:\Windows\inf\setupapi.dev.log"
+        driver_logs = params.get("driver_debug_file", debug_files).split(",")
+        result_dir = test.resultsdir
+
+        for log_file in driver_logs:
+            logging.info("Copy debug file %s from guest to host" % log_file)
+            try:
+                vm.copy_files_from(log_file, result_dir)
+            except FileTransferNotFoundError:
+                logging.error("Didn't find file %s in guest" % log_file)
+
+    def get_driver_from_cdrom(driver_name, session):
+        """
+        Get driver path from virtio-win.iso, and driver version from inf file
+ 
+        :param driver_name: name of the driver
+        return: driver path and driver version
+        """
+        error.context("Get driver driver path and driver version"
+                       " from virtio-win iso", logging.info)
+        device_key = params.get("device_key", "VolumeName like '%virtio-win%'")
+        guest_name_alias = params.get("guest_alias")
+        guest_name = params.get("guest_name")
+        guest_list = {}
+        for i in guest_name_alias.split(","):
+            guest, guest_alias = i.split(":")
+            guest_list[guest] = guest_alias
+        guest_n = guest_list[guest_name]
+        virtio_win_disk = utils_misc.get_win_disk_vol(session, device_key) 
+        if not virtio_win_disk:
+            raise error.Error("Didn't fine virtio-win iso in guest")
+
+        driver_path = "%s:\%s\%s" % (virtio_win_disk, driver_name, guest_n)
+        cmd = "dir /B %s\*.inf" % driver_path
+        status, output = session.cmd_status_output(cmd) 
+        if status:
+            raise error.TestError("%s" % output)
```

Comment:
there may be one or more blank space after version in .inf file， so convert it to int
e.g "8600   " != "8600"